modules/gghealthd/src/sd_bus.c

```diff
@@ -317,3 +319,64 @@ GglError get_lifecycle_state(
     err = get_component_result(bus, unit_path, state);
     return err;
 }
+
+void reset_restart_counters(sd_bus *bus, const char *qualified_name) {
```

Comment:
nit: In general I think it's better to return errors when possible, and let the caller of the function decide whether to ignore and continue running or not. e.g. if systemd isn't responding we might want to retry this a couple times before continuing on. Not a big deal in this case though